sympy/functions/combinatorial/factorials.py

```diff
@@ -851,12 +857,17 @@ def eval(cls, n, k):
         elif d.is_zero is False:
```

Comment:
I do not understand why this should be checked.

Reply:
Okay, there is this case `binomial(n, -1)` where n is integer, this has two answers, if n = -1 the result should be `1` and if other than -1 it should yield `0`, so since there is ambiguity here `binomial(n, -1)` should result as it is, but if we remove `d.is_zero is False` it would yield 0. This is where this condition comes to play

Reply:
I think that `binomial(n, -1)` should be 0. I suspect that this condition is some kind of relic from an old version.

Reply:
okay this is in accordance with what you said earlier about all negative integer k regardless of n. I should remove the condition then.